core/src/main/scala/org/apache/spark/scheduler/cluster/CoarseGrainedSchedulerBackend.scala

```diff
@@ -88,6 +89,12 @@ class CoarseGrainedSchedulerBackend(scheduler: TaskSchedulerImpl, val rpcEnv: Rp
   @GuardedBy("CoarseGrainedSchedulerBackend.this")
   private val executorsPendingToRemove = new HashMap[String, Boolean]
 
+  // Mark executors that we will request to kill in the near future.
+  // This is different from executors in executorsPendingToRemove, which have already asked to be
+  // killed.
+  @GuardedBy("CoarseGrainedSchedulerBackend.this")
+  private val executorsToBeKilled = mutable.Set.empty[String]
```

Comment:
Instead of introducing yet another state for executors, would it be possible to re-use the `disableExecutor` logic for this? Sounds like that mostly maps to what you want - an executor that is not available for scheduling but still hasn't fully been decommissioned.

Reply:
I removed executorsToBeKilled. I tried using ```disableExecutor``` but it had the side effect of immediately removing executors. Instead I reused the pre-existing ```executorsPendingToBeRemoved``` map, and added a new parameter to ```killExecutors``` to get the behavior I need.